packages/@aws-cdk/aws-vpclattice-alpha/lib/service.ts

```diff
@@ -0,0 +1,220 @@
+   * @returns a token representing the service ARN
+   */
+  readonly serviceArn: string;
+
+  /**
+   * The VPC lattice service ID
+   *
+   * @attribute
+   * @returns a token representing the service ID
+   */
+  readonly serviceId: string;
+
+  /**
+   * The DNS entry for the VPC lattice service
+   */
+  readonly dnsEntry?: DnsEntry;
+}
+
+/**
+ * Base class for VPC Lattice services
+ */
+abstract class ServiceBase extends Resource implements IService {
+  /**
+   * The VPC lattice service ARN
+   */
+  public abstract readonly serviceArn: string;
+
+  /**
+   * The VPC lattice service ID
+   */
+  public abstract readonly serviceId: string;
+
+  /**
+   * The DNS entry for the VPC lattice service
+   */
+  public readonly dnsEntry?: DnsEntry;
+}
+
+/**
+ * Attributes for importing an existing VPC Lattice service
+ */
+export interface ServiceAttributes {
+  /**
+   * The VPC lattice service ARN
+   */
+  readonly serviceArn: string;
+
+  /**
+   * The VPC lattice service ID
+   */
+  // TODO: restore ID? Constructing the ARN from the ID will need accountId, region, etc
+  // readonly serviceId: string;
+}
+
+/**
+ * A VPC Lattice service
+ */
+export class Service extends ServiceBase {
+  /**
+   * Import an existing VPC Lattice service into this CDK application.
+   *
+   * @param scope the parent creating construct (usually `this`)
+   * @param id the construct's name // TODO
+   * @param attrs attributes of the service
+   * @returns An object representing the imported VPC Lattice service.
+   */
+  public static fromServiceAttributes(scope: Construct, id: string, attrs: ServiceAttributes): IService {
+    class UnownedServiceReference extends ServiceBase {
+      public readonly serviceArn: string = attrs.serviceArn;
+      // Extract the serviceId from the ARN, which has the format `arn:aws:vpc-lattice:<region>:<account>:service/<service-id>`
+      public readonly serviceId: string = Stack.of(scope).splitArn(attrs.serviceArn, ArnFormat.SLASH_RESOURCE_NAME).resourceName!;
+    }
+    return new UnownedServiceReference(scope, id);
+  }
+
+  /**
+   * The ARN of the VPC Lattice service
+   * @attribute
+   * @returns a token representing the service ARN
+   */
+  public readonly serviceArn: string;
+
+  /**
+   * The ID of the VPC Lattice service
+   * @attribute
+   * @returns a token representing the service ID
+   */
+  public readonly serviceId: string;
+
+  /**
+   * The DNS entry for the VPC Lattice service
+   */
+  public readonly dnsEntry?: DnsEntry;
+
+  /**
+   * Creates a new VPC Lattice service
+   *
+   * @param scope The parent construct
+   * @param id The construct ID
+   * @param props The service properties
+   */
+  constructor(scope: Construct, id: string, props: ServiceProps) {
+    super(scope, id);
+
+    if (props.authType === ServiceAuthType.AWS_IAM && props.authPolicy === undefined) {
+      throw new ValidationError('An auth policy must be provided when the authType is `AWS_IAM`', this);
+    }
```

Comment:
this tells me that this should be a single property with a more complex object:

```ts
  /**
   * The authentication type for the service
   *
   * @default ServiceAuth.NONE
   */
  readonly auth: ServiceAuth;
```

```ts
class ServiceAuth {
  public static NONE = new ServiceAuth('NONE');
  public static iam(policy: PolicyDocument): ServiceAuth {
    return new ServiceAuth('IAM', {
      authPolicy: policy,
    );
  }

  private constructor() { ... }
}
```

There are plenty of examples around this in the code base. But needs a better example in the design guidelines
https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md#enums